### PR TITLE
(Closes #2665) improve error messages for line-length limiting and kernel arguments.

### LINKED ISSUE #2665: PSyclone line length limit error message prints wrong filename
I came across this yesterday and not sure if it's a me thing! I have an [algorithm](https://code.metoffice.gov.uk/trac/lfric_apps/browser/main/branches/dev/dimitriostheodorakis/vn1.0_cape_cin/science/physics/source/aviation/algorithms/cape_cin_mixed_layer_alg_mod.X90), cape_cin_mixed_layer_alg_mod.X90, in LFRic that I tried to add another [kernel](https://code.metoffice.gov.uk/trac/lfric_apps/browser/main/branches/dev/dimitriostheodorakis/vn1.0_cape_cin/science/physics/source/aviation/kernels/init_ml_and_sl_parcel_ascent_kernel_mod.F90) to. When I then tried to build lfric_atm using the build system I get the following:

```shell
16:48:45 [1mPreprocessing[0m aviation/algorithms/cape_cin_mixed_layer_alg_mod.X90
16:48:45 [1mPreprocessing[0m aviation/algorithms/aviation_diags_alg_mod.X90
16:48:45 [1mPSyclone - global optimisation[0m /tmp/persistent/vn1.0_cape_cin/applications/lfric_atm/working/build_lfric_atm/aviation/algorithms/cape_cin_mixed_layer_alg_mod.x90
16:48:45 [1mPSyclone - global optimisation[0m /tmp/persistent/vn1.0_cape_cin/applications/lfric_atm/working/build_lfric_atm/aviation/algorithms/aviation_diags_alg_mod.x90
Warning: Algorithm Error: Algorithm file contains no invoke() calls: refusing to generate empty PSy code
Parse Error: the file does not conform to the specified 132 line length limit
```

Which is because the kernel contains a line with 134 characters. I was under the impression that we ignored any line limit in the build system and am not sure if this is an fparser or PSyclone Parse Error. Is there a way of printing with the error message the actual file it is complaining about? (In this case the kernel when only the algorithms are mentioned in the output)

[Link to the rev of my branch](https://code.metoffice.gov.uk/trac/lfric_apps/browser/main/branches/dev/dimitriostheodorakis/vn1.0_cape_cin?rev=2602) where this is occuring.

### PR DESCRIPTION
**Pull request by arporter**

### CODE REVIEW

**Comment by arporter** [comment]
A small change that improves some error messages (related to line-length limiting and kernel-argument checks). One for @hiker, @TeranIvy, @oakleybrunt or @mo-lottieturner.

File: src/psyclone/dynamo0p3.py [hunk 1]
```diff
@@ -4970,15 +4971,18 @@ def descriptors(self):
...
+    kernel metadata.
 
     :param call: the object produced by the parser that describes the
                  kernel call to be checked.
     :type call: :py:class:`psyclone.parse.algorithm.KernelCall`
+    :param parent_call: the kernel-call object.
+    :type parent_call: :py:class:`psyclone.domain.lfric.LFRicKern`
+
     :raises: GenerationError if the kernel arguments in the Algorithm layer
              do not match up with the kernel metadata
```
Comment (oakleybrunt) [inline src/psyclone/dynamo0p3.py:4987]: End with a full-stop, please.

File: src/psyclone/dynamo0p3.py [hunk 2]
```diff
@@ -5042,7 +5051,7 @@ class DynKernelArguments(Arguments):
     :type call: :py:class:`psyclone.parse.KernelCall`
     :param parent_call: the kernel-call object.
     :type parent_call: :py:class:`psyclone.domain.lfric.LFRicKern`
-    :param bool check: whether to check for consistency between the \
+    :param bool check: whether to check for consistency between the
         kernel metadata and the algorithm layer. Defaults to True.
```
Comment (oakleybrunt) [inline src/psyclone/dynamo0p3.py:5055]: Could you align this with the above line please?

File: src/psyclone/parse/utils.py
Comment (oakleybrunt) [inline src/psyclone/parse/utils.py:1]: pylint reports:

```
src/psyclone/parse/utils.py:97:8: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)
src/psyclone/parse/utils.py:123:8: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)
src/psyclone/parse/utils.py:129:8: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)
```

**Review by oakleybrunt — CHANGES_REQUESTED** [review]
All the changes look great, much more informative :) Just 3 style changes, very minor.

File: src/psyclone/dynamo0p3.py (hunk 1 above)
Comment (arporter) [reply src/psyclone/dynamo0p3.py:4987]: Done.

File: src/psyclone/dynamo0p3.py (hunk 2 above)
Comment (arporter) [reply src/psyclone/dynamo0p3.py:5055]: This isn't mandated by the style guide - you either indent by (a multiple of) 4 spaces *or* line it up with the previous line, whichever you prefer. In this case I've done the former to save having to split to another line.

File: src/psyclone/parse/utils.py
Comment (arporter) [reply src/psyclone/parse/utils.py:1]: Oh yes, done.

**Comment by arporter** [comment]
CI and Integration Tests permitting, this is ready for another look now.

**Review by oakleybrunt — APPROVED** [review]
Thank you for making the changes :) I can't merge this PR so I suppose I'm giving you permission?

**Comment by arporter** [comment]
> Thank you for making the changes :) I can't merge this PR so I suppose I'm giving you permission?

I've given you the power now! You'll need to follow the steps at:
https://github.com/stfc/PSyclone/wiki/CodeReview#merging-a-branch-to-master

**Pull request merged**